src/main/java/net/minecraftforge/common/capabilities/CapabilityDispatcher.java

```diff
@@ -101,15 +101,17 @@ public CapabilityDispatcher(Map<ResourceLocation, ICapabilityProvider> list, Lis
         {
             LazyOptional<T> ret = c.getCapability(cap, side);
             //noinspection ConstantConditions
-            if (ret == null) {
+            if (ret == null)
+            {
                 throw new RuntimeException(
                         String.format(
                                 "Provider %s.getCapability() returned null; this is not allowed!",
```

Comment:
I'd also recommend adding something like "return LazyOptional.empty() instead" to the existing error message.

Reply:
Other than that, looks good to me.

Reply:
Just crashing is fine. We need to stop bad mods as soon as possible.

Reply:
Erm, I don't understand.

It already 'just crashes'.

Crashing with a better error message was pretty much the entire point of this PR. I suggested adding more clarification to the error message....

Furthermore, it used to be convention to return null. Documenting the fact that LazyOptional.empty() should now be returned instead will likely stop at least some unnecessary questions.

Reply:
Updated the message.